File: leetcode/wordPattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
 var wordPattern = function(pattern, s) {
  let map1 = new Map()
  let map2 = new Map()
  
  let sSplit = s.split(" ")
  
  if(pattern.length != sSplit.length) return false
  
  for(let i = 0; i < pattern.length; i++) {
      if(map1.has(pattern[i])) {
          if(map1.get(pattern[i]) != sSplit[i]) return false
      }else map1.set(pattern[i], sSplit[i])
      
      if(map2.has(sSplit[i])) {
          if(map2.get(sSplit[i]) != pattern[i]) return false
      }else map2.set(sSplit[i], pattern[i])
  }
  
  return true
};

File: leetcode/wordPattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        sSplit = s.split(' ')
        
        if(len(pattern) != len(sSplit)): return False
        
        map1 = {}
        map2 = {}
        
        for i in range(len(pattern)):
            if pattern[i] in map1:
                if map1[pattern[i]] != sSplit[i]: return False
            else: map1[pattern[i]] = sSplit[i]
            
            if sSplit[i] in map2:
                if map2[sSplit[i]] != pattern[i]: return False
            else: map2[sSplit[i]] = pattern[i]
                
        return True
                